Doc/c-api/intro.rst

```diff
@@ -251,24 +251,31 @@ they are done with the result; this soon becomes second nature.
 Reference Count Details
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-The reference count behavior of functions in the Python/C API is best  explained
+The reference count behavior of functions in the Python/C API is best explained
 in terms of *ownership of references*.  Ownership pertains to references, never
 to objects (objects are not owned: they are always shared).  "Owning a
-reference" means being responsible for calling Py_DECREF on it when the
+reference" means being responsible for calling :c:func:`Py_DECREF` on it when the
 reference is no longer needed.  Ownership can also be transferred, meaning that
 the code that receives ownership of the reference then becomes responsible for
 eventually decref'ing it by calling :c:func:`Py_DECREF` or :c:func:`Py_XDECREF`
 when it's no longer needed---or passing on this responsibility (usually to its
-caller). When a function passes ownership of a reference on to its caller, the
-caller is said to receive a *new* reference.  When no ownership is transferred,
-the caller is said to *borrow* the reference. Nothing needs to be done for a
+caller). Let's look at different types of reference ownership passings in more
+detail.
+
+When a function returns an object and effectively increases the
+reference count of it, the function is said to *give* ownership of a new
+reference to its caller and the caller is said to *own* the reference. When a
+function returns an object without changing the reference count of it, the
+caller is said to *borrow* the reference. Nothing needs to be done for a
 borrowed reference.
 
-Conversely, when a calling function passes in a reference to an  object, there
-are two possibilities: the function *steals* a  reference to the object, or it
-does not.  *Stealing a reference* means that when you pass a reference to a
-function, that function assumes that it now owns that reference, and you are not
-responsible for it any longer.
+Conversely, when a calling function passes in a reference to an object, there
+are two possibilities: the function *steals* a reference to the object, or it
+does not. If a called function decreases the reference count of an object, it
```

Comment:
This is not clear to me or is directly wrong. The called function has two choices:

1) Increase the refcount - take ownership
2) Don't increase the refcount - stealing ownership

In any case the function never decreases the reference count.

Reply:
Thank you for the correction🙏 I will rewrite the sentences after your reply for the previous thread.

Reply:
I am really sorry for the hassle and I'm afraid I can't fully understand what you are saying here. Basically, I get what you mean. "Stealing a reference does not mean or guarantee decreasing in refcount". I will change the doc accordingly.
But please note that, for example, `PyTuple_SetItem` does decrease refcount of an passed-in object in some cases.